### ISSUE TITLE
Redundant Self : self not remove in specific case

### ISSUE THREAD

**Issue by VinceBurn** [nicklockwood/SwiftFormat#752]
Hi @nicklockwood 

I've found that `self` is not remove before an optional binding shadowing

This code with only: rule "redudantSelf", option "remove"
```
func redundantSelf(aString: String,
                   aInt: Int,
                   aBool: Bool) {
    self.count = 4
    self.title = aString
    self.textLabel.text = String(aInt)
    let foo = self.optional
    print(foo ?? -1)
    self.optional = nil

    guard let optional = self.optional else { return }
    print(optional)
}
```

will produce
```
func redundantSelf(aString: String,
                   aInt: Int,
                   aBool: Bool) {
    count = 4
    title = aString
    textLabel.text = String(aInt)
    let foo = self.optional
    print(foo ?? -1)
    self.optional = nil

    guard let optional = self.optional else { return }
    print(optional)
}
```

We can notice the `self.optional` before the `guard` statement.

This feels like a very low priority issue.

**Comment by nicklockwood** [comment]
@VinceBurn thanks for reporting - I'll look into it.

**Comment by nicklockwood** [comment]
@VinceBurn took a while, but this is fixed in 0.48.12
